packages/vm/src/imports.rs

```diff
@@ -1938,42 +2126,44 @@ mod tests {
     #[cfg(feature = "iterator")]
     fn do_db_next_works() {
         let api = MockApi::default();
-        let (env, _instance) = make_instance(api);
+        let (fe, mut store, _instance) = make_instance(api);
+        let mut fe_mut = fe.into_mut(&mut store);
 
-        leave_default_data(&env);
+        leave_default_data(&mut fe_mut);
 
-        let id = do_db_scan(&env, 0, 0, Order::Ascending.into()).unwrap();
+        let id = do_db_scan(fe_mut.as_mut(), 0, 0, Order::Ascending.into()).unwrap();
 
         // Entry 1
-        let kv_region_ptr = do_db_next(&env, id).unwrap();
+        let kv_region_ptr = do_db_next(fe_mut.as_mut(), id).unwrap();
         assert_eq!(
-            force_read(&env, kv_region_ptr),
+            force_read(&mut fe_mut, kv_region_ptr),
             [KEY1, b"\0\0\0\x03", VALUE1, b"\0\0\0\x06"].concat()
         );
 
         // Entry 2
-        let kv_region_ptr = do_db_next(&env, id).unwrap();
+        let kv_region_ptr = do_db_next(fe_mut.as_mut(), id).unwrap();
         assert_eq!(
-            force_read(&env, kv_region_ptr),
+            force_read(&mut fe_mut, kv_region_ptr),
             [KEY2, b"\0\0\0\x04", VALUE2, b"\0\0\0\x05"].concat()
         );
 
         // End
-        let kv_region_ptr = do_db_next(&env, id).unwrap();
-        assert_eq!(force_read(&env, kv_region_ptr), b"\0\0\0\0\0\0\0\0");
+        let kv_region_ptr = do_db_next(fe_mut.as_mut(), id).unwrap();
+        assert_eq!(force_read(&mut fe_mut, kv_region_ptr), b"\0\0\0\0\0\0\0\0");
         // API makes no guarantees for value_ptr in this case
     }
 
     #[test]
     #[cfg(feature = "iterator")]
     fn do_db_next_fails_for_non_existent_id() {
         let api = MockApi::default();
-        let (env, _instance) = make_instance(api);
+        let (fe, mut store, _instance) = make_instance(api);
+        let mut fe_mut = fe.into_mut(&mut store);
 
-        leave_default_data(&env);
+        leave_default_data(&mut fe_mut);
 
         let non_existent_id = 42u32;
-        let result = do_db_next(&env, non_existent_id);
+        let result = do_db_next(fe_mut.as_mut(), non_existent_id);
```

Comment:
Like with `clone()`, seems some of these `as_mut()` can be removed. Not exactly sure which ones.

Reply:
yeah, you are right. They create short living semi-clones.